mbedtls-sys/build/config.rs

```diff
@@ -36,24 +36,53 @@ pub const PREFIX: &'static str = r#"
 #endif
 "#;
 
+/*
+
+DEFAULT_DEFINES was generated using the following script
+
+#!/usr/bin/python
+
+import re
+
+simple_define = re.compile(r'(.*)#define (MBEDTLS_[A-Z0-9_]+)$')
+define_with_default = re.compile(r'.*#define (MBEDTLS_[A-Z0-9_]+) +([0-9A-Za-z_]+)')
+
+def format(macro, state):
+    return "    (\"%s\", %s)," % (macro, state.rjust(49 - len(macro) + len(state)))
+
+for line in open('vendor/include/mbedtls/config.h').readlines():
+    match = simple_define.match(line)
+
+    if match:
+        state = "Undefined" if match.group(1) == '//' else "Defined"
+        print format(match.group(2), state)
+    else:
+        match = define_with_default.match(line)
+        if match:
+            print format(match.group(1), "Undefined") + (" // default: %s" % (match.group(2)))
+*/
+
 #[cfg_attr(rustfmt, rustfmt_skip)]
 pub const DEFAULT_DEFINES: &'static [CDefine] = &[
+    ("MBEDTLS_CONFIG_H",                                  Defined),
     ("MBEDTLS_HAVE_ASM",                                  Defined),
-    ("MBEDTLS_NO_64BIT_MULTIPLICATION",                   Undefined),
     ("MBEDTLS_NO_UDBL_DIVISION",                          Undefined),
+    ("MBEDTLS_NO_64BIT_MULTIPLICATION",                   Undefined),
     ("MBEDTLS_HAVE_SSE2",                                 Undefined),
-    ("MBEDTLS_HAVE_TIME",                                 Undefined),
-    ("MBEDTLS_HAVE_TIME_DATE",                            Undefined),
+    ("MBEDTLS_HAVE_TIME",                                 Defined),
+    ("MBEDTLS_HAVE_TIME_DATE",                            Defined),
```

Comment:
We definitely don't want this change

Reply:
If I understand correctly, you mean edit vendor config.h and then regenerate config.rs? Or just edit config.rs and leave the vendor version untouched?

It is not clear to me if config.rs is supposed to reflect the contents of config.h or to potentially override it.

Reply:
The `config.h` in `vendor/` is not used. The `config.h` is generated at build time from `config.rs`. All defines in `config.h` should be represented in `config.rs`, but the values may differ to suit our needs.